src/resources/files.ts

```diff
@@ -103,14 +107,40 @@ export interface FilesAPI {
   update: (changes: FileChangeUpdate[]) => Promise<FilesMetadata[]>;
 }
 
+function waitUntilFileIsUploaded(
+  fileId: CogniteInternalId,
+  axiosInstance: AxiosInstance,
+  resourcePath: string,
+  frequencyInMs: number = 1000
+) {
+  return new Promise((resolve, reject) => {
+    try {
+      const retrieve = generateRetrieveSingleEndpoint<
+        CogniteInternalId,
+        FilesMetadata
+      >(axiosInstance, resourcePath, new MetadataMap());
+
+      const myInterval = setInterval(async () => {
+        if ((await retrieve(fileId)).uploaded) {
+          clearInterval(myInterval);
```

Comment:
will it wait forever?
what if status is `failed`?

Reply:
So status can never be failed? But the upload process can fail, but that is catched before this function should be called.

Reply:
hmm ok